Todor efcore ds (#2168)
telerik/reporting-docs: designing-reports/connecting-to-data/data-source-components/entitycoredatasource-component/overview.md

---
title: Overview
page_title: EntityCoreDataSource Component at a Glance
description: "Learn how the EntityCoreDataSource component in Telerik Reporting binds reports to an Entity Framework Core DbContext at design time and runtime."
slug: entitycoredatasource-overview
tags: entitycoredatasource,efcore,entityframeworkcore,datasource,dbcontext,overview
tag: new
published: True
position: 0
reportingArea: General
---

# EntityCoreDataSource Component Overview

> The `EntityCoreDataSource` component was introduced with [2026 Q2 (20.1.26.520)](https://www.telerik.com/support/whats-new/reporting/release-history/progress-telerik-reporting-2026-q2-(20-1-26-520)). It connects report data items to an [Entity Framework Core](https://learn.microsoft.com/en-us/ef/core/) `DbContext`. It exposes business objects defined in EF Core models to report authors directly, without intermediate object data sources or hand-written code in code-behind. The toolbox label for the component is **Entity Framework Core Data Source**, and it is available in the [Standalone Report Designer (.NET)](slug:telerikreporting/designing-reports/report-designer-tools/desktop-designers/standalone-report-designer/overview) and the [Web Report Designer](slug:telerikreporting/designing-reports/report-designer-tools/web-report-designer/overview).

Reviewer: ```suggestion
> The EntityCoreDataSource component was introduced with [2026 Q2 (20.1.26.520)](https://www.telerik.com/support/whats-new/reporting/release-history/progress-telerik-reporting-2026-q2-(20-1-26-520)).

The **EntityCoreDataSource** component allows you to connect report data items to an [Entity Framework Core](https://learn.microsoft.com/en-us/ef/core/) `DbContext`. It exposes business objects defined in EF Core models directly, without intermediate object data sources or hand-written code in code-behind.

The component is available both in the [Standalone Report Designer (.NET)](slug:telerikreporting/designing-reports/report-designer-tools/desktop-designers/standalone-report-designer/overview) and the [Web Report Designer](slug:telerikreporting/designing-reports/report-designer-tools/web-report-designer/overview) under the toolbox label **Entity Framework Core Data Source**.
```

I suggest moving the introduction info as a note, as after some time passes, this will become no longer a highlight. The intro needs to be specific about what the component allows the end user to achieve.

> The `EntityCoreDataSource` component derives from `EntityDataSourceBase`, which it shares with its .NET Framework 4.6.2 equivalent, the [EntityDataSource](slug:telerikreporting/designing-reports/connecting-to-data/data-source-components/entitydatasource-component/overview). Both components expose the same `Context`, `ContextMember`, `ConnectionString`, and `Parameters` surface; the EF Core variant targets `Microsoft.EntityFrameworkCore.DbContext` derivatives, whereas the old variant targets the .NET Framework `ObjectContext` and EF6 `DbContext`.

Reviewer: I'd move this info as a note too, and remove some of the technical explanations that do not affect the end users (could be mistaken, please correct me if the end user actually needs to know some of the technical details explained here).

```suggestion
> If your project targets .NET Framework, use the [EntityDataSource](slug:telerikreporting/designing-reports/connecting-to-data/data-source-components/entitydatasource-component/overview) component instead. If you are migrating from EntityDataSource, the two components expose the same `Context`, `ContextMember`, `ConnectionString`, and `Parameters` properties, so your existing configuration carries over.
```

## Key Capabilities

> - **Dedicated design-time support:** the component has its own toolbox item, design-time editor, and a configuration wizard that is shared with the [EntityDataSource Wizard](slug:telerikreporting/designing-reports/report-designer-tools/desktop-designers/tools/data-source-wizards/entitydatasource-wizard). It populates the [Data Explorer](slug:telerikreporting/designing-reports/report-designer-tools/desktop-designers/tools/data-explorer) with the schema of the selected entity and supports live data preview.

Reviewer: Not sure what "shared with the EntityDataSource" and "design-time editor" mean here. Isn't design-time editor and wizard the same thing?

Maybe change to something like "**Dedicated design-time support**: the component includes a dedicated toolbox item and a wizard in both the Standalone Report Designer (.NET) and the Web Report Designer. The [wizard](slug:telerikreporting/designing-reports/report-designer-tools/desktop-designers/tools/data-source-wizards/entitydatasource-wizard) guides you through selecting a `DbContext`, choosing the entity member to expose as the data source, and configuring any parameters. After the wizard completes, the [Data Explorer](slug:telerikreporting/designing-reports/report-designer-tools/desktop-designers/tools/data-explorer) shows the entity schema so you can drag fields directly onto report items. The last wizard step lets you preview live data before finishing."

> - **`DbContext` binding:** assign a `DbContext` type or instance to the `Context` property and the name of a `DbSet<T>` or queryable property to the `ContextMember` property; the component retrieves the data automatically when the report is processed.

Reviewer: If I understand correctly, `Context`/`ContextMember` only affect scenarios where the EFCore component is created with code. Maybe change to something like:

```suggestion
- **`DbContext` binding:** set the data source to a `DbContext` and specify which entity set, property, or method provides the report data. The wizard handles this through a selection UI; in code, set `Context` and `ContextMember` directly or use the relevant constructor.
```

> - **Connection-string injection:** assign a connection string to the `ConnectionString` property of the data source, or pass it through one of the constructor overloads. The configured value is forwarded to a `DbContext` constructor that accepts a `string`.

Reviewer: Same here:

```suggestion
- **Connection-string injection:** configure which database connection the data source uses. Set it through the wizard, via the `ConnectionString` property in code, or as the first argument of the `EntityCoreDataSource(connectionString, context, contextMember)` constructor.
```

- **Parameter passing:** add entries to the `Parameters` collection to invoke parameterized methods or queryable projections on the `DbContext`.

> - **Shared Data Source reuse:** promote an `EntityCoreDataSource` instance to a [Shared Data Source](slug:telerikreporting/designing-reports/connecting-to-data/data-source-components/shareddatasource-component) and reuse the same configuration across multiple reports.

Reviewer: ```suggestion
- **Shared Data Source reuse:** in the Web Report Designer, extract the created `EntityCoreDataSource` component to a [Shared Data Source](slug:telerikreporting/designing-reports/connecting-to-data/data-source-components/shareddatasource-component) and reuse the same configuration across multiple reports.
```

## Public API

The component exposes three constructor overloads and three serializable properties:

```CSharp
namespace Telerik.Reporting
{
    public class EntityCoreDataSource : EntityDataSourceBase
    {
        public EntityCoreDataSource();
        public EntityCoreDataSource(object context, string contextMember);
        public EntityCoreDataSource(string connectionString, object context, string contextMember);

        // Inherited from EntityDataSourceBase:
        public string ConnectionString { get; set; }
        public object Context { get; set; }            // DbContext type or instance
        public string ContextMember { get; set; }      // DbSet<T> or method/property name
    }
}
```

> The `Parameters` collection is inherited through `ObjectDataSourceBase` and accepts the standard fluent `Add(name, type, valueOrExpression)` overload used by every Telerik Reporting data source.

Reviewer: ```suggestion
The `Parameters` collection is available through the inherited `ObjectDataSourceBase` class. Use `Add(name, type, valueOrExpression)` to add entries.
```

## DbContext Requirements

To consume a `DbContext` with the `EntityCoreDataSource` component, the type must satisfy the following requirements:

> - Your Entity Framework Core (EFCore) assembly should be allowed and accessible by the Report Designer along with all its dependencies. For the custom EFCore assembly, you need to [Extend the Report Designer to Recognize Custom Assemblies](slug:telerikreporting/designing-reports/report-designer-tools/desktop-designers/standalone-report-designer/configuration/extending-report-designer). Its dependencies, though, must be copied manually to the report designer's folder. You may take them from the `bin` folder of the corresponding project after building it.

Reviewer: ```suggestion
- Your Entity Framework Core (EFCore) assembly must be allowed and accessible by the Report Designer along with all its dependencies. For the custom EFCore assembly, you need to [extend the report designer to recognize custom assemblies](slug:telerikreporting/designing-reports/report-designer-tools/desktop-designers/standalone-report-designer/configuration/extending-report-designer). Its dependencies, though, must be copied manually to the report designer's folder. You can take them from the `bin` folder of the corresponding project after building it.
```

> - Your Entity Framework Core (EFCore) assembly should be allowed and accessible by the Report Designer along with all its dependencies. For the custom EFCore assembly, you need to [Extend the Report Designer to Recognize Custom Assemblies](slug:telerikreporting/designing-reports/report-designer-tools/desktop-designers/standalone-report-designer/configuration/extending-report-designer). Its dependencies, though, must be copied manually to the report designer's folder. You may take them from the `bin` folder of the corresponding project after building it.

Reviewer: ```suggestion
- Your Entity Framework Core (EF Core) assembly should be allowed and accessible by the Report Designer along with all its dependencies. For the custom EFCore assembly, you need to [Extend the Report Designer to Recognize Custom Assemblies](slug:telerikreporting/designing-reports/report-designer-tools/desktop-designers/standalone-report-designer/configuration/extending-report-designer). Its dependencies, though, must be copied manually to the report designer's folder. You may take them from the `bin` folder of the corresponding project after building it.
```

EFCore > EF Core for consistency with the other places

Reviewer: Overall, I would consider merging point 1 (about discoverability in SRD.NET) and 2 (about discoverability in project) into a single point, and including the WRD scenario as well; something like:

- **Assembly accessibility**: the assembly that contains your `DbContext` must be accessible wherever the report runs:
  - **Your application (runtime)**: add the `DbContext` class library as a project reference and register it under `telerikReporting.assemblyReferences` in `appsettings.json`.
  - **Standalone Report Designer (.NET)**: extend the designer to recognize the assembly. The EF Core runtime and database provider assemblies must be copied manually to the designer folder. You can take them from the `bin` folder of the project after building it.
  - **Web Report Designer**: add the assembly to the host application and register it under `telerikReporting.assemblyReferences` in `appsettings.json`.

> 	> note It is a work in progress from our side to improve this behavior.

Reviewer: ```suggestion
	> note Automatic resolution of transitive dependencies in the report designer is planned for a future release. Until then, copy the required assemblies manually.
```
